common/rocksdb_env_s3.h

```diff
@@ -54,7 +53,12 @@ class S3FatalException : public std::exception {
 class S3Env : public Env {
```

Comment:
Can you document the design, assumption (and why), and limitation (and why) of the implementation?

Reply:
sure. Added more details.